src/TLuaInterpreterMapper.cpp

```diff
@@ -577,18 +577,30 @@ int TLuaInterpreter::addRoom(lua_State* L)
     const int id = getVerifiedInt(L, __func__, 1, "roomID");
     const Host& host = getHostFromLua(L);
     const bool added = host.mpMap->addRoom(id);
-    lua_pushboolean(L, added);
```

Comment:
This line was what was causing the problem as it was being placed on the stack before the `getVerifiedInt(...)` could read the areaID from it.